api/src/main/java/com/code4ro/nextdoor/authentication/controller/AuthenticationController.java

```diff
@@ -51,15 +53,25 @@ public AuthenticationController(final AuthenticationService authenticationServic
         return ResponseEntity.created(location).build();
     }
 
-    @ApiOperation(value = "Login user in application")
+    @ApiOperation("Login user in application")
     @PostMapping("/login")
     public ResponseEntity<JwtAuthenticationResponse> login(@Valid @RequestBody final LoginRequest loginRequest) {
         final Authentication authentication = authenticationManager.authenticate(
                 new UsernamePasswordAuthenticationToken(loginRequest.getEmail(), loginRequest.getPassword()));
 
         SecurityContextHolder.getContext().setAuthentication(authentication);
 
-        final String jwt = tokenProvider.generateToken(authentication);
-        return ResponseEntity.ok(new JwtAuthenticationResponse(jwt));
+        final UserPrincipal userPrincipal = (UserPrincipal) authentication.getPrincipal();
```

Comment:
Why is a custom implementation for this and not a normal authentication flow?
Default SpringSecurity Authentication using JWT is super simple to implement without extra work needed.
And for the long run is the way to go